infrahub_sdk/transforms.py

```diff
@@ -25,33 +27,49 @@ class InfrahubTransform:
     query: str
     timeout: int = 10
 
-    def __init__(self, branch: str = "", root_directory: str = "", server_url: str = ""):
+    def __init__(
+        self,
+        branch: str = "",
+        root_directory: str = "",
+        server_url: str = "",
+        client: Optional[InfrahubClient] = None,
+        repository_config: Optional[InfrahubRepositoryConfig] = None,
+    ):
         self.git: Repo
 
         self.branch = branch
-
         self.server_url = server_url or os.environ.get("INFRAHUB_URL", "http://127.0.0.1:8000")
         self.root_directory = root_directory or os.getcwd()
+        self.repository_config = repository_config
 
-        self.client: InfrahubClient
+        self._client = client
 
         if not self.name:
             self.name = self.__class__.__name__
 
         if not self.query:
             raise ValueError("A query must be provided")
 
+    @property
+    def client(self) -> InfrahubClient:
+        if not self._client:
+            self._client = InfrahubClient(address=self.server_url)
+
+        return self._client
+
     @classmethod
     async def init(cls, client: Optional[InfrahubClient] = None, *args: Any, **kwargs: Any) -> InfrahubTransform:
         """Async init method, If an existing InfrahubClient client hasn't been provided, one will be created automatically."""
+        warnings.warn(
+            "InfrahubClient.init has been deprecated and will be removed in Infrahub SDK 0.14.0 or the next major version",
```

Comment:
Sorry about the delay around reviewing this will take a closer look soon! Just a quick note about this which looks like a typo. Even if it's a minor detail this shouldn't be `InfrahubClient.init` it should be either `InfrahubTransform.init` or `{cls.__class__.__name__}.init`